src/renderer/components/chat/ChatList.tsx

```diff
@@ -67,34 +136,164 @@ export default function ChatList(props: {
+                        }}
+                      />
+                    </div>
+                  )
+                }}
+              </ChatListPart>
+              {isSearchActive && (
+                <>
+                  <div
+                    className='search-result-divider'
+                    style={{ width: width }}
+                  >
+                    {translate_n('n_contacts', contactIds.length)}
+                  </div>
+                  <ChatListPart
+                    isRowLoaded={isContactLoaded}
+                    loadMoreRows={loadContact}
+                    rowCount={contactIds.length}
+                    width={width}
+                    height={contactsHeight(height)}
+                  >
+                    {({ index, key, style }) => {
+                      const contactId = contactIds[index]
+                      return (
+                        <div key={key} style={style}>
+                          {contactCache[contactId] ? (
+                            <ContactListItem
+                              contact={contactCache[contactId]}
+                              showCheckbox={false}
+                              checked={false}
+                              showRemove={false}
+                              onClick={async _ => {
+                                let chatId = await DeltaBackend.call(
+                                  'contacts.getChatIdByContactId',
+                                  contactId
+                                )
+                                onChatClick(chatId)
+                              }}
+                            />
+                          ) : (
+                            <PlaceholderChatListItem />
+                          )}
+                        </div>
+                      )
+                    }}
+                  </ChatListPart>
+                  {chatListIds.length > 0 ||
+                    PseudoListItemAddContact({
+                      queryStr,
+                      queryStrIsEmail,
+                      onClick: addContactOnClick,
+                    })}
+                  <div
+                    className='search-result-divider'
+                    style={{ width: width }}
+                  >
+                    {translate_n('n_messages', messageResultIds.length)}
+                  </div>
+
+                  <ChatListPart
+                    isRowLoaded={isMessageLoaded}
+                    loadMoreRows={loadMessages}
+                    rowCount={messageResultIds.length}
+                    width={width}
+                    height={
+                      // take remaining space
+                      messagesHeight(height)
+                    }
+                  >
+                    {({ index, key, style }) => {
+                      const msrId = messageResultIds[index]
+                      return (
+                        <div style={style} key={key}>
```

Comment:
Can we factor this out in an own function component? The function is already huuuge

Reply:
I don't think that makes much sense here.